### ISSUE TITLE
Sending data after error (custom url)

### ISSUE THREAD

**Issue by klap14** [mendhak/gpslogger#895]
Hi, is it possible to send data after server error? When the app cant send data to my web it will wait for example 10 minutes and then try it again? Now if there is error i lost data. Thanks

**Comment by mendhak** [comment]
Do you mean an HTTP 500 error code?  While technically it could be possible to get the app to retry on such a code, 10 minutes would be too much as the queue to send Custom URL can grow massively if the user has a fast logging frequency and especially when doze mode kicks in, which is a real problem at the moment, so this would exacerbate confusion.  

Additional to that, if there is a server error, there is still no guarantee that the error goes away after 10 minutes or even an hour if the server fix requires manual intervention and the owner isn't aware of it.  So adding more time would be an unreliable bandaid.   

I do see that the Custom URL feature has an exponential backoff of 5 seconds. 

https://github.com/mendhak/gpslogger/blob/master/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/customurl/CustomUrlJob.java#L101

Maybe this could be increased to 30 seconds to give it a little more grace period?  But how useful is 30 seconds.

**Comment by klap14** [comment]
I rather thought the app would repeat it several times. If the server goes down from the provider side and the site has e.g. a 2 hour outage, for the app to upload all the data after the outage (so it will try to do it in some interval). Of course, if there is a lot of data, it will take a long time, but if the data is written to a file, the application could pull it out and only after the succesfull upload (no error returned) would mark it as uploaded and not repeat the attempts.

**Comment by andreapx** [comment]
I'm having the same problem because my smartphone lately have troubles with wifi, so it often fails to upload to ownCloud, but I don't know about it until I check the files on Nextcloud server.
@mendhak it would be nice to have the possibility to force the upload of the logs at a specific time, or to have an intent that I can call with Tasker.

**Comment by mendhak** [comment]
Related to Custom URL, in v119 there's now the ability to upload to Custom URL.  It uses the CSV logging feature to replay requests to the target server.  This way if some uploads have failed  you can rerun the Custom URL requests.

**Comment by andreapx** [comment]
I've been using the custom URL since years, I don't get what is he new ability.
When it fails the upload, I click on the "UP ARROW" in the bottom, I select "ownCloud" and select the last days to be sure.
